homeassistant/components/radarr/config_flow.py

```diff
@@ -51,29 +56,28 @@ async def async_step_reauth_confirm(
         self._set_confirm_only()
         return self.async_show_form(step_id="reauth_confirm")
 
-    async def async_step_user(
+    async def async_step_user(self, _: dict[str, Any] | None = None) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        if is_hassio(self.hass):
+            for addon in get_addons_info(self.hass).values():
+                if DOMAIN in addon[ATTR_SLUG]:
+                    self._url = addon["network"]["7878/tcp"]
+                    return await self.async_step_on_supervisor()
```

Comment:
This is not how supervisor discovery works.

Instead, the supervisor should announce / push the add-on being installed.

For reference implementations, please see adguard, mqtt or deconz.

Reply:
Had a discussion with @balloob on Discord. He says we won't get discovery messages. That's only ones we've allowed currently.

Reply:
Right, so this is working around it? We should not add workarounds.